hack/conformance/check_conformance_test_requirements.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This tool is for checking conformance e2e tests follow the requirements
+// which is https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	//e.g. framework.ConformanceIt("should provide secure master service ", func() {
+	patternStartConformance = "framework.ConformanceIt\\(.*, func\\(\\) {$"
+	patternEndConformance   = "}\\)$"
+	patternSkipProviderIs   = "Skip.*ProviderIs\\("
+)
+
+var exitCode = 0
+
+// This function checks the requirement: it works for all providers (e.g., no SkipIfProviderIs/SkipUnlessProviderIs calls)
+func checkAllProviders(e2eFile string) error {
+	inConformanceCode := false
+
+	regStartConformance := regexp.MustCompile(patternStartConformance)
+	regEndConformance := regexp.MustCompile(patternEndConformance)
+	regSkipProviderIs := regexp.MustCompile(patternSkipProviderIs)
+
+	fileInput, err := ioutil.ReadFile(e2eFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to read file %s: %v\n", e2eFile, err)
```

Comment:
you could use errors.Wrap here and just output on the calling function.

Reply:
Thanks, I've learn that is useful today.
Done